pkg/cluster/generateconfig_test.go

```diff
@@ -0,0 +1,207 @@
+						{
+							Name:       "default",
+							Visibility: api.VisibilityPublic,
+							IP:         "1.2.3.4",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	for _, tt := range []generateConfigTest{
+		{
+			name: "valid document generates valid installconfig",
+			fixture: func(f *testdb.Fixture) {
+				f.AddOpenShiftClusterDocuments(genGood())
+			},
+			checker: func(ic *installconfig.InstallConfig) []string {
+				var errs []string
+				errs = append(errs, deep.Equal(ic.Config.ObjectMeta.Name, "cluster")...)
+				errs = append(errs, deep.Equal(ic.Config.BaseDomain, "example.aroapp.io")...)
+				return errs
+			},
+		},
+		{
+			name: "incorrect version causes error",
+			fixture: func(f *testdb.Fixture) {
+				doc := genGood()
+				doc.OpenShiftCluster.Properties.ClusterProfile.Version = "v3.11"
+				f.AddOpenShiftClusterDocuments(doc)
+			},
+			wantErr: `unimplemented version "v3.11"`,
+		},
+		{
+			name: "incomplete domains have the environment base domain appended",
+			fixture: func(f *testdb.Fixture) {
+				doc := genGood()
+				doc.OpenShiftCluster.Properties.ClusterProfile.Domain = "notadomain"
+				f.AddOpenShiftClusterDocuments(doc)
+			},
+			checker: func(ic *installconfig.InstallConfig) []string {
+				var errs []string
+				errs = append(errs, deep.Equal(ic.Config.ObjectMeta.Name, "notadomain")...)
+				errs = append(errs, deep.Equal(ic.Config.BaseDomain, "aroapp.io")...)
+				return errs
+			},
+		},
+		{
+			name: "installer cluster validation failure causes error",
+			fixture: func(f *testdb.Fixture) {
+				doc := genGood()
+				doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret = "{}"
+				f.AddOpenShiftClusterDocuments(doc)
+			},
+			wantErr: `pullSecret: Invalid value: "{}": auths required`,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			_, log := testlog.New()
+
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+			_env := mock_env.NewMockInterface(controller)
+			_env.EXPECT().DeploymentMode().AnyTimes().Return(deployment.Development)
+			_env.EXPECT().Zones(gomock.Any()).AnyTimes().Return([]string{"useast2a"}, nil)
+			_env.EXPECT().ACRDomain().AnyTimes().Return("example.com")
+			_env.EXPECT().Domain().AnyTimes().Return("aroapp.io")
+
+			dbOpenShiftClusters, _ := testdb.NewFakeOpenShiftClusters()
+			dbSubscriptions, _ := testdb.NewFakeSubscriptions()
+
+			f := testdb.NewFixture().WithOpenShiftClusters(dbOpenShiftClusters).WithSubscriptions(dbSubscriptions)
+			tt.fixture(f)
+			err := f.Create()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			doc, err := dbOpenShiftClusters.Get(ctx, strings.ToLower(resourceID))
+			if err != nil {
+				t.Error(err)
+			}
+
+			c := &manager{
+				log: log,
+				doc: doc,
+				env: _env,
+			}
+
+			installConfig, platformCreds, image, err := c.generateInstallConfig(ctx)
+
+			if tt.wantErr == "" {
+				if err != nil {
+					t.Error(err)
+				}
+				if installConfig == nil {
+					t.Error(installConfig)
```

Comment:
Might want to consider exiting here with `t.Fatal()`. Otherwise, if `installConfig == nil`, the test will panic at line 201:

```
go test ./pkg/cluster/... -run="TestBackendTry"         
--- FAIL: TestBackendTry (0.00s)                                                                                                                  
    --- FAIL: TestBackendTry/valid_document_generates_valid_installconfig (0.00s)                                                                 
        generateconfig_test.go:181: <nil>                                                                                                         
panic: runtime error: invalid memory address or nil pointer dereference [recovered]                                                               
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]                                                       
        panic: runtime error: invalid memory address or nil pointer dereference                                                                   
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2d419b9]                                                                           
                                                                                                                                                  
goroutine 83 [running]:                                                                                                                           
testing.tRunner.func1.1(0x3350240, 0x5f5c2c0)                                                                                                     
        /usr/local/go/src/testing/testing.go:999 +0x319                                                                                           
testing.tRunner.func1(0xc00079e7e0)                                                                                                               
        /usr/local/go/src/testing/testing.go:1002 +0x402                                                                                          
panic(0x3350240, 0x5f5c2c0)                                                                                                                       
        /usr/local/go/src/runtime/panic.go:969 +0x166                                                                                             
github.com/golang/mock/gomock.(*Controller).Finish(0xc0007a8540)
        /home/isim/workspace/go/src/github.com/Azure/ARO-RP/vendor/github.com/golang/mock/gomock/controller.go:246 +0x27a
panic(0x3350240, 0x5f5c2c0)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/Azure/ARO-RP/pkg/cluster.TestBackendTry.func3(0x0, 0xc0008cdd60, 0x1, 0x1)
        /home/isim/workspace/go/src/github.com/Azure/ARO-RP/pkg/cluster/generateconfig_test.go:102 +0x29
github.com/Azure/ARO-RP/pkg/cluster.TestBackendTry.func8(0xc00079e7e0)
        /home/isim/workspace/go/src/github.com/Azure/ARO-RP/pkg/cluster/generateconfig_test.go:201 +0x871
testing.tRunner(0xc00079e7e0, 0xc000212b80)
        /usr/local/go/src/testing/testing.go:1050 +0xdc
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1095 +0x28b
FAIL    github.com/Azure/ARO-RP/pkg/cluster     0.022s
```